provider/cluster/kube/workload_builder.go

```diff
@@ -0,0 +1,184 @@
+
+	"github.com/ovrclk/akash/manifest"
+	clusterUtil "github.com/ovrclk/akash/provider/cluster/util"
+	mtypes "github.com/ovrclk/akash/x/market/types"
+)
+
+const (
+	StorageAttributePersistent = "persistent"
+	StorageAttributeClass      = "class"
+	StorageAttributeMount      = "mount"
+	StorageAttributeReadOnly   = "readOnly"
+)
+
+type workloadBuilder struct {
+	builder
+	service          *manifest.Service
+	runtimeClassName string
+}
+
+func newWorkloadBuilder(log log.Logger, settings Settings, lid mtypes.LeaseID, group *manifest.Group, service *manifest.Service) workloadBuilder {
+	return workloadBuilder{
+		builder: builder{
+			settings: settings,
+			log:      log.With("module", "kube-builder"),
+			lid:      lid,
+			group:    group,
+		},
+		service:          service,
+		runtimeClassName: settings.DeploymentRuntimeClass,
+	}
+}
+
+func (b *workloadBuilder) container() corev1.Container {
+	falseValue := false
+
+	kcontainer := corev1.Container{
+		Name:    b.service.Name,
+		Image:   b.service.Image,
+		Command: b.service.Command,
+		Args:    b.service.Args,
+		Resources: corev1.ResourceRequirements{
+			Limits:   make(corev1.ResourceList),
+			Requests: make(corev1.ResourceList),
+		},
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		SecurityContext: &corev1.SecurityContext{
+			RunAsNonRoot:             &falseValue,
+			Privileged:               &falseValue,
+			AllowPrivilegeEscalation: &falseValue,
+		},
+	}
+
+	if cpu := b.service.Resources.CPU; cpu != nil {
+		requestedCPU := clusterUtil.ComputeCommittedResources(b.settings.CPUCommitLevel, cpu.Units)
+		kcontainer.Resources.Requests[corev1.ResourceCPU] = resource.NewScaledQuantity(int64(requestedCPU.Value()), resource.Milli).DeepCopy()
+		kcontainer.Resources.Limits[corev1.ResourceCPU] = resource.NewScaledQuantity(int64(cpu.Units.Value()), resource.Milli).DeepCopy()
+	}
+
+	if mem := b.service.Resources.Memory; mem != nil {
+		requestedMem := clusterUtil.ComputeCommittedResources(b.settings.MemoryCommitLevel, mem.Quantity)
+		kcontainer.Resources.Requests[corev1.ResourceMemory] = resource.NewQuantity(int64(requestedMem.Value()), resource.DecimalSI).DeepCopy()
+		kcontainer.Resources.Limits[corev1.ResourceMemory] = resource.NewQuantity(int64(mem.Quantity.Value()), resource.DecimalSI).DeepCopy()
+	}
+
+	for _, storage := range b.service.Resources.Storage {
+		attr := storage.Attributes.Find(StorageAttributePersistent)
+		if persistent, _ := attr.AsBool(); persistent {
+			continue
+		}
+
+		requestedStorage := clusterUtil.ComputeCommittedResources(b.settings.StorageCommitLevel, storage.Quantity)
+		kcontainer.Resources.Requests[corev1.ResourceEphemeralStorage] = resource.NewQuantity(int64(requestedStorage.Value()), resource.DecimalSI).DeepCopy()
+		kcontainer.Resources.Limits[corev1.ResourceEphemeralStorage] = resource.NewQuantity(int64(storage.Quantity.Value()), resource.DecimalSI).DeepCopy()
+	}
+
+	// TODO: this prevents over-subscription. skip for now.
```

Comment:
I think this comment is obsolete at this point. I have no idea what it means anymore